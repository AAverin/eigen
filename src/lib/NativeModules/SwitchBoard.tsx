import { findNodeHandle, NativeModules } from "react-native"
const { ARSwitchBoardModule } = NativeModules

function presentNavigationViewController(component: React.Component<any, any>, route: string) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.presentNavigationViewController(reactTag, route)
}

function presentModalViewController(component: React.Component<any, any>, route: string) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.presentModalViewController(reactTag, route)
}

function dismissModalViewController(component: React.Component<any, any>) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.dismissModalViewController(reactTag)
}

export default {
  presentNavigationViewController,
  presentModalViewController,
  dismissModalViewController,
}
