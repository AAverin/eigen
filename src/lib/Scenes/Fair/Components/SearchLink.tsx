import { BorderBox, Box, Sans, Spacer } from "@artsy/palette"
import SwitchBoard from "lib/NativeModules/SwitchBoard"
import React from "react"
import { Image, TouchableWithoutFeedback } from "react-native"
import styled from "styled-components/native"

const Icon = styled(Image)`
  width: 20;
  height: 20;
`

interface Props {
  fairID: string
}

export class SearchLink extends React.Component<Props> {
  handlePress = () => {
    const { fairID } = this.props
    SwitchBoard.presentNavigationViewController(this, `/${fairID}/search`)
  }

  render() {
    return (
      <Box>
        <TouchableWithoutFeedback onPress={this.handlePress}>
          <BorderBox justifyContent="center" alignItems="center" flexDirection="row" p={1}>
            <Icon source={require("../../../../../images/SearchButton.png")} />
            <Spacer width={12} />
            <Sans size="3" weight="medium">
              Find Exhibitors and Artists
            </Sans>
          </BorderBox>
        </TouchableWithoutFeedback>
      </Box>
    )
  }
}
