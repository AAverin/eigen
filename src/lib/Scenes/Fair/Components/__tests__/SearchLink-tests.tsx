import { shallow } from "enzyme"
import React from "react"
import * as renderer from "react-test-renderer"

jest.mock("lib/NativeModules/SwitchBoard", () => ({
  presentNavigationViewController: jest.fn(),
}))

import SwitchBoard from "lib/NativeModules/SwitchBoard"
import { SearchLink } from "../SearchLink"

describe("SearchLink", () => {
  it("Renders properly", () => {
    const comp = renderer.create(<SearchLink id="this-is-a-fair-id" _id="123456" />)
    expect(comp).toMatchSnapshot()
  })

  it("Routes to Fair Search on click", () => {
    const comp = shallow(<SearchLink id="this-is-a-fair-id" _id="123456" />)
    const inst = comp.instance()

    inst.handlePress()

    expect(SwitchBoard.presentNavigationViewController).toHaveBeenCalledWith(inst, "/this-is-a-fair-id/search")
  })
})
