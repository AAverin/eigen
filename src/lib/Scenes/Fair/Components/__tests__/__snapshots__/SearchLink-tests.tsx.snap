// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLink Renders properly 1`] = `
<View
  mt={1}
  style={
    Array [
      Object {
        "marginTop": 4,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    alignItems="center"
    flexDirection="row"
    height={48}
    justifyContent="center"
    mb={1}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    p={1}
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "marginBottom": 4,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
        Array [
          Object {
            "borderColor": "#E5E5E5",
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 48,
            "marginBottom": 4,
            "paddingBottom": 4,
            "paddingLeft": 4,
            "paddingRight": 4,
            "paddingTop": 4,
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../images/SearchButton.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "width": 12,
          },
          undefined,
        ]
      }
      width={12}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      className="sc-bdVaJa bDWFJH"
      ellipsizeMode="tail"
      fontFamily="Unica77LL-Medium"
      fontSize="12px"
      lineHeight="16px"
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 12,
            "lineHeight": 16,
          },
          undefined,
        ]
      }
    >
      Find Exhibitors and Artists
    </Text>
  </View>
</View>
`;
